libs/SalesforceSDKCore/SalesforceSDKCoreTests/SalesforceSDKManagerTests.m

```diff
@@ -98,50 +116,91 @@ - (void)testOneLaunchAtATime
 {
     [self createStandardPostLaunchBlock];
     [self createTestAppIdentity];
-    
     [self launchAndVerify:YES failMessage:@"Initial launch attempt should have been successful."];
     [self launchAndVerify:NO failMessage:@"Second concurrent launch should not be allowed."];
+    [self verifyPostLaunchState];
+}
+
+- (void)testOverrideAiltnAppNameBeforeSDKManagerLaunch
+{
+    NSString *prevName = SalesforceSDKManager.ailtnAppName;
+    [SalesforceSDKManager setAiltnAppName:kTestAppName];
+    [self createStandardPostLaunchBlock];
+    [self createTestAppIdentity];
+    [self launchAndVerify:YES failMessage:@"Launch attempt should have been successful."];
+    [self verifyPostLaunchState];
+    [self compareAiltnAppNames:kTestAppName];
+    [SalesforceSDKManager setAiltnAppName:prevName];
+}
+
+- (void)testOverrideAiltnAppNameAfterSDKManagerInit
+{
+    NSString *prevName = SalesforceSDKManager.ailtnAppName;
+    [self createStandardPostLaunchBlock];
+    [self createTestAppIdentity];
+    [self launchAndVerify:YES failMessage:@"Launch attempt should have been successful."];
+    [self verifyPostLaunchState];
+    [SalesforceSDKManager setAiltnAppName:kTestAppName];
+    [self compareAiltnAppNames:kTestAppName];
+    [SalesforceSDKManager setAiltnAppName:prevName];
+}
+
+- (void)testDefaultAiltnAppName
+{
+    NSString *prevName = SalesforceSDKManager.ailtnAppName;
+    [self createStandardPostLaunchBlock];
+    [self createTestAppIdentity];
+    [self launchAndVerify:YES failMessage:@"Launch attempt should have been successful."];
+    [self verifyPostLaunchState];
+    NSString *appName = [[NSBundle mainBundle] infoDictionary][(NSString *) kCFBundleNameKey];
+    [self compareAiltnAppNames:appName];
+    [SalesforceSDKManager setAiltnAppName:prevName];
     
+}
+
+- (void)testOverrideInvalidAiltnAppName
+{
+    NSString *prevName = SalesforceSDKManager.ailtnAppName;
+    [SFUserAccountManager sharedInstance].currentUser = [self createUserAccount];
```

Comment:
Oh, the saving and rehydrating already happens in setUp and tearDown for this class.  But I notice that most of the places that are using `createUserAccount` are not cleaning up the physical account after.

Perhaps `restoreOrigSdkManagerState` should be amended to check for the existence of a current user (which otherwise is `nil`ed out before every test), and if it exists, remove the account before resetting the current user back to the original value.